### PR TITLE
feat: track modifier for examples

### PR DESCRIPTION
**Pull request by andiwand**
while dealing with vertexing issues I thought adding a track parameter modification algorithm to the examples framework might be useful

another try of https://github.com/acts-project/acts/pull/1658

### CODE REVIEW

File: Examples/Algorithms/TruthTracking/ActsExamples/TruthTracking/TrackModifier.cpp [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    {
+      auto& params = trk.parameters();
+
+      if (m_cfg.killTime) {
+        params[Acts::eBoundTime] = 0;
+      }
+    }
+
+    {
+      auto& optCov = trk.covariance();
```
Comment (benjaminhuth) [inline Examples/Algorithms/TruthTracking/ActsExamples/TruthTracking/TrackModifier.cpp:54]: same as above

File: Examples/Algorithms/TruthTracking/ActsExamples/TruthTracking/TrackModifier.cpp [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  if (m_cfg.inputTrajectories.empty() != m_cfg.outputTrajectories.empty()) {
+    throw std::invalid_argument(
+        "Input and output for trajectories and track parameters have to be "
+        "used consistently");
+  }
+}
+
+ActsExamples::ProcessCode ActsExamples::TrackModifier::execute(
+    const ActsExamples::AlgorithmContext& ctx) const {
+  auto modifyTrack = [this](auto trk) {
```
Comment (benjaminhuth) [inline Examples/Algorithms/TruthTracking/ActsExamples/TruthTracking/TrackModifier.cpp:42]: I think we should decide if we make a copy in the function parameter, or explicit in the line below

File: Examples/Algorithms/TruthTracking/ActsExamples/TruthTracking/TrackModifier.cpp [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    auto newTrk = trk;
+
+    {
+      auto& params = trk.parameters();
```
[the remark this answers is not in the document]
Comment (andiwand) [reply Examples/Algorithms/TruthTracking/ActsExamples/TruthTracking/TrackModifier.cpp:46]: nice catch!

**Review by benjaminhuth — CHANGES_REQUESTED** [review]
Looks good, only some things in the modifier-lambda are not entirely clear to me yet

**Review by benjaminhuth — APPROVED** [review]
Let's go!

**Pull request merged**